observability-lib/dashboards/core-node-components/component_test.go

```diff
@@ -11,12 +12,51 @@ import (
 	corenodecomponents "github.com/smartcontractkit/chainlink-common/observability-lib/dashboards/core-node-components"
 )
 
+var generate = flag.Bool("generate", false, "generate test output")
+var fileOutput = "test-output.json"
+
+func TestGenerateFile(t *testing.T) {
+	if *generate == false {
+		t.Skip("skipping test")
+	}
+
+	testDashboard, err := corenodecomponents.NewDashboard(&corenodecomponents.Props{
+		Name:              "Core Node Components Dashboard",
+		MetricsDataSource: grafana.NewDataSource("Prometheus", ""),
+		LogsDataSource:    grafana.NewDataSource("Loki", ""),
+		Tested:            true,
+	})
+	if err != nil {
+		t.Errorf("Error creating dashboard: %v", err)
+	}
+	json, errJSON := testDashboard.GenerateJSON()
+	if errJSON != nil {
+		t.Errorf("Error generating JSON: %v", errJSON)
+	}
+	if _, errExists := os.Stat(fileOutput); errExists == nil {
+		errRemove := os.Remove(fileOutput)
+		if errRemove != nil {
+			t.Errorf("Error removing file: %v", errRemove)
+		}
+	}
+	file, errFile := os.Create(fileOutput)
+	if errFile != nil {
+		panic(errFile)
+	}
+	defer file.Close()
```

Comment:
could also consider using `t.Cleanup`. 

Have you considered making `TestGenerateFile` part of `TestNewDashboard`? IFIUC, both tests have the same input and output when calling `NewDashboard` and `GenerateJSON`

Reply:
NewDashboard inputs are different per dashboard 
the file generated has different content per dashboard

Reply:
`TestGenerateFile` :

```

	testDashboard, err := corenodecomponents.NewDashboard(&corenodecomponents.Props{
		Name:              "Core Node Components Dashboard",
		MetricsDataSource: grafana.NewDataSource("Prometheus", ""),
		LogsDataSource:    grafana.NewDataSource("Loki", ""),
		Tested:            true,
	})
	if err != nil {
		t.Errorf("Error creating dashboard: %v", err)
	}
	json, errJSON := testDashboard.GenerateJSON()
	if errJSON != nil {
		t.Errorf("Error generating JSON: %v", errJSON)
	}
```

`TestNewDashboard`: 
```
func TestNewDashboard(t *testing.T) {
	t.Run("NewDashboard creates a dashboard", func(t *testing.T) {
		testDashboard, err := corenodecomponents.NewDashboard(&corenodecomponents.Props{
			Name:              "Core Node Components Dashboard",
			MetricsDataSource: grafana.NewDataSource("Prometheus", ""),
			LogsDataSource:    grafana.NewDataSource("Loki", ""),
			Tested:            true,
		})
		if err != nil {
			t.Errorf("Error creating dashboard: %v", err)
		}
		require.IsType(t, grafana.Dashboard{}, *testDashboard)
		require.Equal(t, "Core Node Components Dashboard", *testDashboard.Dashboard.Title)
		json, errJSON := testDashboard.GenerateJSON()
		if errJSON != nil {
			t.Errorf("Error generating JSON: %v", errJSON)
		}
```

look equivalent to me? what am i missing?